File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0153_Find_Minimum_in_Rotated_Sorted_Array.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0153_Find_Minimum_in_Rotated_Sorted_Array {
    public int findMin(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int l = 0, r = nums.length - 1, m;
        while (l < r) {
            m = l + (r - l) / 2;
            if (nums[m] < nums[r]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return nums[r];
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0153_Find_Minimum_in_Rotated_Sorted_Array.cc
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
 public:
  int findMin(vector<int>& nums) {
    if (nums.empty()) {
      return 0;
    }
    int l = 0, r = nums.size() - 1, m;
    while (l < r) {
      m = l + (r - l) / 2;
      if (nums[m] < nums[r]) {
        r = m;
      } else {
        l = m + 1;
      }
    }
    return nums[l];
  }
};
